python/ray/util/spark/cluster_init.py

```diff
@@ -1615,11 +1615,11 @@ def ray_cluster_job_mapper(_):
         )
         job_rdd = job_rdd.withResources(resource_profile)
 
-    job_rdd.mapPartitions(ray_cluster_job_mapper).collect()
-
     hook_entry = _create_hook_entry(is_global=(ray_temp_dir is None))
     hook_entry.on_spark_job_created(spark_job_group_id)
```

Comment:
Move this line to front of `job_rdd.mapPartitions(ray_cluster_job_mapper).collect()` because this function runs inside background thread and `job_rdd.mapPartitions(ray_cluster_job_mapper).collect()` will block forever until cluster is terminated.